src/reflect/scala/reflect/internal/tpe/FindMembers.scala

```diff
@@ -59,6 +59,29 @@ trait FindMembers {
       result
     }
 
+    /*--- Refinement classes and parents  **/
+    private[this] var refinedClasses: List[Symbol] = Nil
+    private[this] var refinedParents: List[List[Symbol]] = Nil
+
+    /* Returns true if any of the classes has the cl class as a parent.
+     * The process fetches and stores each class lists of parents, as needed */
+    private[this] def isParentOfAnyRefinementClass(cl: Symbol): Boolean = {
+      // 1 first, try the parentSymbols it already has
+      var found = refinedParents.exists(_.contains(cl))
+      // 2 then, fetch and store the parents from the classes try
+      while (!found && ! refinedClasses.isEmpty) {
+        refinedClasses match {
+          case hh :: tt =>
+            refinedClasses = tt
+            val pp = hh.parentSymbols
+            refinedParents ::= pp
+            found = pp.contains(cl)
+          case Nil => // should not occur
```

Comment:
I think it would be simpler, and just as quick, to instead use:

`refinedClasses.exists(_.info.parents.exists(_.typeSymbol == cl)))`

This basically inlines `def parentSymbols = info.parents.map(_.typeSymbol)`, avoiding the list creation.

The two closures in that expression should be compiled to be statically hoisted instances by `LambdaMetafactory`, so that doesn't add allocations.